File: problems/009/009.lua
posix = require "posix"

local function f()
    local perimeter = 1000

    for c = math.floor(perimeter / 3) + 1, math.floor(perimeter / 2) do
        for b = math.floor((perimeter - c) / 2) + 1, c do
            a = perimeter - b -c

            if a * a + b * b == c * c then
                return a * b * c
            end
        end
    end
end

function toNs(ts)
    return ts[1] * 1000000000 + ts[2]
end

if #arg == 0 then
    print(f())
elseif #arg == 1 then
    iters = tonumber(arg[1])
    start = { posix.clock_gettime("monotonic") }
    for i = 1,iters do
        f()
    end
    end_ = { posix.clock_gettime("monotonic") }
    print(toNs(end_) - toNs(start))
end

File: problems/009/009.js
function f() {
    var perimeter = 1000;
    var a, b, c;

    for (c = ((perimeter / 3) | 0) + 1; c < (perimeter / 2) | 0; c++) {
        for (b = (((perimeter - c) / 2) | 0) + 1; b < c; b++) {
            a = perimeter - b - c;

            if (a * a + b * b === c * c) {
                return a * b * c;
            }
        }
    }

    return 0;
}

function toNs(ts) {
    return ts.sec * 1000000000 + ts.nsec;
}

var clock = require('posix-clock');

if (process.argv.length === 3) {
    var i;
    var iters = parseInt(process.argv[2], 10);

    var start = clock.gettime(clock.MONOTONIC);
    for (i = 0; i < iters; i++) {
        f();
    }
    var end = clock.gettime(clock.MONOTONIC);

    console.log(toNs(end) - toNs(start));
} else {
    console.log(f());
}
